### PR TITLE
Adding information to help track down remote authentication issues

### PR DESCRIPTION
**Pull request by golharam**
This should help others track down issues when using external authentication.

### CODE REVIEW

File: lib/galaxy/web/framework/middleware/remoteuser.py [hunk 1]
```diff
@@ -143,6 +145,10 @@ def __call__( self, environ, start_response ):
             # The API handles its own authentication via keys
             return self.app( environ, start_response )
         else:
+            log.debug("Unable to identify user.  %s not found" % self.remote_user_header)
+            for key in environ.keys():
+                log.debug("%s = %s" % (key, environ.get(key)))
```
Comment (nsoranzo) [inline lib/galaxy/web/framework/middleware/remoteuser.py]: I think this is better:

```
            for k, v in environ.iteritems():
                log.debug("%s = %s" % (k, v))
```

**Comment by jmchilton** [comment]
:+1:

File: lib/galaxy/web/framework/middleware/remoteuser.py (hunk 1 above)
Comment (blankenberg) [inline lib/galaxy/web/framework/middleware/remoteuser.py]: For log, it is best to not do the string substitution manually (with %), but to provide it as bare arguments to the log.xyz function: `log.debug( "%s = %s", k, v )`

**Comment by golharam** [comment]
Ok, I updated the patch

**Comment by nsoranzo** [comment]
Thanks @golharam , I've created a small PR for your branch to fix the flake8 error.

**Comment by martenson** [comment]
@galaxybot test this

**Comment by martenson** [comment]
@galaxybot test this

**Comment by martenson** [comment]
@galaxybot ok to test

**Comment by jmchilton** [comment]
@erasche `minor` tag has a specific meeting for release notes generation, it should never be added to a PR that comes from outside the committer group -  these PRs no matter how small need to be mentioned so the opener gets credit for the contribution.

**Comment by hexylena** [comment]
@jmchilton understood. That was not previously clear. I took minor to mean "really minor changes." Will use it more appropriately.

**Comment by martenson** [comment]
@golharam please merge https://github.com/golharam/galaxy/pull/1 and I will merge this PR. Thank you.

**Comment by golharam** [comment]
I just did....

**Comment by hexylena** [comment]
Great, thanks @golharam! As soon as the tests pass, we'll get this merged. Thank you for your contribution.

**Pull request merged**